v0/destinations/klaviyo/transform.js

```diff
@@ -160,11 +160,85 @@ const trackRequestHandler = (message, category, destination) => {
-  if (destination.Config.enforceEmailAsPrimary) {
-    delete customerProperties.$id;
-    customerProperties._id = getFieldValueFromMessage(message, "userId");
+  return customerProperties;
+};
+const trackRequestHandler = (message, category, destination) => {
+  let payload = {};
+  let event = get(message, "event");
+  event = event ? event.trim().toLowerCase() : event;
+  if (ecomEvents.includes(event) && message.properties) {
+    const eventName = eventNameMapping[event];
+    payload.event = eventName;
+    payload.token = destination.Config.publicApiKey;
+    const eventMap = jsonNameMapping[eventName];
+    // using identify to create customer properties
+    payload.customer_properties = createCustomerProperties(message);
+    if (
+      !payload.customer_properties.$email &&
+      !payload.customer_properties.$phone_number
+    ) {
+      throw new CustomError(
+        "email or phone is required for customer_properties",
+        400
+      );
+    }
+    const categ = CONFIG_CATEGORIES[eventMap];
+    payload.properties = constructPayload(
+      message.properties,
+      MAPPING_CONFIG[categ.name]
+    );
+
+    // products mapping using Items.json
+    if (message.properties.items && Array.isArray(message.properties.items)) {
+      const itemArr = [];
+      message.properties.items.forEach(key => {
+        let item = constructPayload(
+          key,
+          MAPPING_CONFIG[CONFIG_CATEGORIES.ITEMS.name]
+        );
+        item = removeUndefinedAndNullValues(item);
+        if (!isEmptyObject(item)) {
+          itemArr.push(item);
+        }
+      });
+      if (!payload.properties) {
+        payload.properties = {};
+      }
+      payload.properties.items = itemArr;
+    }
+
+    // all extra props passed is incorporated inside properties
+    let customProperties = {};
+    customProperties = extractCustomFields(
+      message,
+      customProperties,
+      ["properties"],
+      ecomExclusionKeys
+    );
+    if (!isEmptyObject(customProperties)) {
+      payload.properties = {
+        ...payload.properties,
+        ...customProperties
+      };
+    }
+
+    if (isEmptyObject(payload.properties)) {
+      delete payload.properties;
+    }
+  } else {
+    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+    payload.token = destination.Config.publicApiKey;
+    if (message.properties && message.properties.revenue) {
+      payload.properties.$value = message.properties.revenue;
+      delete payload.properties.revenue;
+    }
+    const customerProperties = createCustomerProperties(message);
+    if (destination.Config.enforceEmailAsPrimary) {
+      delete customerProperties.$id;
+      customerProperties._id = getFieldValueFromMessage(message, "userId");
+    }
+    payload.customer_properties = customerProperties;
   }
-  payload.customer_properties = customerProperties;
   if (message.timestamp) {
     payload.time = toUnixTimestamp(message.timestamp);
```

Comment:
Let's use `getFieldValueFromMessage()` to extract timestamp and also insert a check whether timestamp is already in unix format otherwise this function will truncate the last 3 digits and contain invalid data.